### DISCUSSION TITLE
如何隐藏某些类型文献的文献类型和载体标识符

### DISCUSSION THREAD

**Discussion by YDX-2147483647** [BITNP/BIThesis#412] [Q&A] [answered]
> [!CAUTION]
>
> 您正在违反国标！
>
> 如果您只是想加奖学金，那没必要放到`*.bib`里，可以单独列出。

> [!TIP]
>
> 如果要删除所有标识符，`\usepackage[…, gbtype=false, …]{biblatex}`或`\settoggle{bbx:gbtype}{false}`即可。

## 一种hack

下面这个例子会删掉`[Z]`。

```latex
\DeclareStyleSourcemap{
  \maps[datatype=bibtex]{
    \map{
      \pertype{misc}
      \step[fieldset=usera, null] % 文献类型和载体标识符
      % usera 代表属于 biblatex-gb7714-2015 内部实现，搜索大写字母Z找到的。
    }
  }
}
```

![GE}2M)_R64_9EUXV X6BSTM](https://github.com/BITNP/BIThesis/assets/73375426/496279c7-2b4a-4ced-a752-9ec549083475)

```bib
@misc{myCiteKey,
  title = {噫吁嚱},
  year = {2006},
}
```

## 参考

https://ctan.org/pkg/biblatex-gb7714-2015

![O8J4PK2W7`D771R$ %H8`PN](https://github.com/BITNP/BIThesis/assets/73375426/3794ba7b-9a3e-40d6-86e8-d6a5f343d199)

https://github.com/hushidong/biblatex-solution-to-latex-bibliography/

![K}GD4MYTJMM)NFA3DA8LK U](https://github.com/BITNP/BIThesis/assets/73375426/feeddede-dab0-4deb-85d5-f7c876d96685)

**Comment by hushidong** [comment, marked as answer]
也可以在局部设置：`\settoggle{bbx:gbtype}{false}`，比如
```
\AtEveryBibitem{\ifentrytype{misc}{\settoggle{bbx:gbtype}{false}}{\settoggle{bbx:gbtype}{true}}}
```
如果不是全局使用，在其内部再增加一些对于环境或者类别的判断，可以让其只适用于某一局部，比如publications中。
